File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    # 고르려는 폰켓몬 마리수
    num = len(nums) // 2
    nums = set(nums)
    
    # 가질 수 있는 종류의 개수를 반환해야한다.
    # 최대는 N / 2
    if len(nums) < num:
        answer = len(nums)
    else:
        answer = num
    
    return answer

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.js
function solution(nums) {
    let answer = 0;
    // N/2마리를 계산
    let num = nums.length / 2;
    
    // 중복 제거
    let new_nums = new Set(nums);
    nums = [...new_nums];
    
    // 만약, 종류 수 < 가져갈 수 있는 폰켓몬수 -> 종류 수
    if (nums.length < num) {
        answer = nums.length;
    } else {
        // 종류 수 > 가져갈 수 있는 폰켓몬수 -> 가져갈 수 있는 폰켓몬 수
        answer = num;
    }
    
    return answer;
}